cpp/src/arrow/flight/server.cc

```diff
@@ -665,16 +665,20 @@ Status FlightServerBase::Init(const FlightServerOptions& options) {
 
   const Location& location = options.location;
   const std::string scheme = location.scheme();
-  if (scheme == kSchemeGrpc || scheme == kSchemeGrpcTcp || scheme == kSchemeGrpcTls) {
+  if (scheme == kSchemeGrpc || scheme == kSchemeGrpcTcp || scheme == kSchemeGrpcTls || scheme == kSchemeGrpcMTls) {
     std::stringstream address;
     address << location.uri_->host() << ':' << location.uri_->port_text();
 
     std::shared_ptr<grpc::ServerCredentials> creds;
-    if (scheme == kSchemeGrpcTls) {
+    if ((scheme == kSchemeGrpcTls) || (scheme == kSchemeGrpcMTls)) {
       grpc::SslServerCredentialsOptions ssl_options;
       for (const auto& pair : options.tls_certificates) {
         ssl_options.pem_key_cert_pairs.push_back({pair.pem_key, pair.pem_cert});
       }
+      if (scheme == kSchemeGrpcMTls) {
```

Comment:
I'd say instead of using a new scheme, just add a flag to `FlightServerOptions`.